### PR TITLE
Added more parts of the spline rendering.

### PR DESCRIPTION
**Pull request by zond**

### CODE REVIEW

File: jxl/src/features/spline.rs [hunk 1]
```diff
@@ -439,7 +445,132 @@ fn for_each_equally_spaced_point<F: FnMut(Point, f32)>(
     f(points[points.len() - 1], accumulated_distance);
 }
 
-const DESIRED_RENDERING_DISTANCE: f32 = 1.0;
+trait Dct32 {
+    fn continuous_idct(&self, t: f32) -> f32;
+}
+
+impl Dct32 for [f32; 32] {
+    fn continuous_idct(&self, t: f32) -> f32 {
```
Comment (veluca93) [inline jxl/src/features/spline.rs:453]: Why is this a trait instead of a free function?

File: jxl/src/features/spline.rs [hunk 2]
```diff
@@ -439,7 +445,132 @@ fn for_each_equally_spaced_point<F: FnMut(Point, f32)>(
...
+        &mut self,
+        spline: &Spline,
+        points_to_draw: &[(Point, f32)],
+        length: f32,
+        segments_by_y: &mut Vec<(u64, usize)>,
+        desired_distance: f32,
+    );
+}
+
+impl SplineSegments for Vec<SplineSegment> {
```
Comment (veluca93) [inline jxl/src/features/spline.rs:514]: Same here, could be two functions, no?

**Review by veluca93 — APPROVED** [review]

File: jxl/src/features/spline.rs (hunk 1 above)
Comment (zond) [reply jxl/src/features/spline.rs:453]: I just like the aesthetic, but I can change it to a function. WDYT?

**Review by mo271 — APPROVED** [review]

File: jxl/src/features/spline.rs (hunk 1 above)
Comment (veluca93) [reply jxl/src/features/spline.rs:453]: I prefer functions unless there's a good reason to go with traits :-)

Comment (zond) [reply jxl/src/features/spline.rs:453]: Done

File: jxl/src/features/spline.rs (hunk 2 above)
Comment (zond) [reply jxl/src/features/spline.rs:512]: Done

**Pull request merged**